Objects/typeobject.c

```diff
@@ -8064,7 +8064,7 @@ static slotdef slotdefs[] = {
     UNSLOT("__abs__", nb_absolute, slot_nb_absolute, wrap_unaryfunc,
            "abs(self)"),
     UNSLOT("__bool__", nb_bool, slot_nb_bool, wrap_inquirypred,
-           "self != 0"),
+           "True if self else False"),
```

Comment:
To me this replacement is nonsensical.  The existing docstring is the docstring for bool, int, float, and complex, all of which have at least one True instance and one False instance.  It was a mistake to use it for None and would be a mistake to replace it with anything that says the same. type(None) has only one instance and its boolean value is false so the appropriate expression is 'False'.
```suggestion
           "False"),
```
Ellipsis and class object have no `.__bool__` methods (and docstrings thereof) because their instance boolean values are True, which is the default for python objects.  (The builtin sequence and collection classes also do not have `.__bool__` methods either, at least not now.  Don't know why.)

Reply:
This isn't the docstring for just None, but the default docstring for all types that implement `__bool__`, including those that do sometimes return False:

```
>>> int.__bool__.__doc__
'self != 0'
>>> float.__bool__.__doc__
'self != 0'
>>> None.__bool__.__doc__
'self != 0'
>>> range.__bool__.__doc__
'self != 0'
```

Your suggestion would be inaccurate for these.

The sequence types don't have a `__bool__` because `bool()` checks `__len__` if `__bool__` doesn't exist. `bool("")` is False because `len("") == 0`.

Reply:
It might be better to have a docstring for just `None.__bool__` that says simply "False", but I believe that it's not possible to create a docstring for a single class's slotted method with the way the slots machinery currently works. And it doesn't seem worth it to rearchitect the slots machinery for such a simple requirement.

Reply:
I already wrote "The existing docstring ['self != 0'] is the [correct] docstring for bool, int, float, and complex".  For those, it is a correct Python version of the C code (leaving aside 0 versus 0.0 versus 0+0j).

For None, that docstring is wrong, in that 'None !=0' is True, not False.  Before posting, I checked (and just rechecked) that the equal strings are not the same strings: "int.__bool__.__doc__ is None.__bool__.__doc__" is False.  I know nothing about the internal machinery but would consider it a bug if different str instances cannot be made different.  Let me test.

Reply:
It creates a new string every time you access the attribute:
```
>>> int.__bool__.__doc__ is int.__bool__.__doc__
False
```